File: day1.py

with open("day1.txt") as f:
    elf = 0
    elves = []
    for line in f:
        if line == "\n":
            elves.append(elf)
            elf = 0
        else:
            elf += int(line)
    elves = sorted(elves)
    print(elves[-1])
    print(sum(elves[-3:]))

File: src/bin/day1.rs
use std::fs::File;
use std::io::{self, BufRead};

fn main() -> io::Result<()> {
    let file = File::open("day1.txt")?;
    let mut elves: Vec<i32> = vec![];
    let mut elf: i32 = 0;
    for line in io::BufReader::new(file).lines()  {
        if let Ok(line) = line {
            if line == "" {
                elves.push(elf);
                elf = 0;
            } else {
                elf += line.parse::<i32>().unwrap()
            }
        }
    }
    elves.sort();
    elves.reverse();
    println!("{}", elves[0]);
    println!("{}", elves[0..3].iter().sum::<i32>());
    Ok(())
}
